lib/model/instance/submission.js

```diff
@@ -36,28 +36,33 @@ const out = {};
 
 // TODO/CR: should this be in its own file for consistency?
 out.PartialSubmission = Instance()(({ Submission }) => class {
-  complete(form, maybeActor) {
+  complete(form, maybeActor, maybeDeviceId) {
```

Comment:
nit:

so, here it's `complete(form: Form, maybeActor: Option[Actor], maybeDeviceId: String?)`.

the `maybeActor` must not be null, but it also must be an `Option`, whereas `maybeDeviceId` may be null. but they're both named `maybe` which is confusing.

i would suggest just naming it `deviceId` and either doing `deviceId = null` to indicate it can be nulled, or else leaving a comment explaining what the types are here.

Reply:
Sounds good! I wasn’t sure how the “maybe” prefix was being used, so I just went with it. I’ll add the “ = null” to indicate it’s nullable.